client/object_put.go

```diff
@@ -168,28 +175,29 @@ func (x *ObjectWriter) WritePayloadChunk(chunk []byte) bool {
 		// the allocated buffer is filled, or when the last chunk is received.
 		// It is mentally assumed that allocating and filling the buffer is better than
 		// synchronous sending, but this needs to be tested.
+		writtenBytes += len(chunk[:ln])
 		x.partChunk.SetChunk(chunk[:ln])
 		x.req.SetVerificationHeader(nil)
 
 		x.err = signServiceMessage(x.signer, &x.req)
 		if x.err != nil {
 			x.err = fmt.Errorf("sign message: %w", x.err)
-			return false
+			return writtenBytes, x.err
```

Comment:
Failed to sign and returning `writtenBytes`?

Reply:
We may already wrote some bytes before, why not?

Reply:
Yes, but this can also be the first message and in this case you'll return some X with 0 actually written.